src/domain/reward.ts

```diff
@@ -81,6 +81,32 @@ export function reputationRedemption(proposalId: Bytes, beneficiary: Address, ti
 }
 
 function shouldRemoveAccountFromUnclaimed(reward: GPReward): boolean {
+  let proposal = getProposal(reward.proposal);
```

Comment:
why need this ? 
use ContributionRewardProposal.load(pid) and check if its not null.